src/composables/pools/usePoolCreation.ts

```diff
@@ -93,19 +102,75 @@ export default function usePoolCreation() {
     poolCreationState.thirdPartyFeeController = address;
   };
 
+  const getScaledAmounts = (): BigNumber[] => {
+    const scaledAmounts: BigNumber[] = poolCreationState.tokenWeights.map(
+      (token: TokenWeight) => {
+        const tokenInfo = getToken(token.tokenAddress);
+        const amount = new BigNumber(token.amount);
+        const scaledAmount = scale(amount, tokenInfo.decimals);
+        return scaledAmount;
+      }
+    );
+    return scaledAmounts;
+  };
+
+  const getPoolSymbol = (): string => {
+    const tokenSymbols = poolCreationState.tokenWeights.map(
+      (token: TokenWeight) => {
+        const weightRounded = Math.round(token.weight);
+        const tokenInfo = getToken(token.tokenAddress);
+        return `${Math.round(weightRounded)}${tokenInfo.symbol}`;
+      }
+    );
+
+    return tokenSymbols.join('-');
+  };
+
+  const createPool = async () => {
+    const provider = getProvider();
+    const poolDetails = await balancerService.pools.create(PoolType.Weighted)(
+      provider,
+      'MyPool',
+      getPoolSymbol(),
+      '0.01',
+      poolCreationState.tokenWeights,
+      // poolCreationState.thirdPartyFeeController
+      AddressZero
+    );
+    poolCreationState.poolId = poolDetails.id;
+    poolCreationState.poolAddress = poolDetails.address;
+  };
+
+  const joinPool = async () => {
+    const provider = getProvider();
+    const response = await balancerService.pools.join(
+      provider,
+      poolCreationState.poolId,
+      account.value,
+      account.value,
+      poolCreationState.tokenWeights,
+      getScaledAmounts()
+    );
+    console.log("Got join pool response: ", response);
+  };
```

Comment:
Because these are submitting transactions we need to handle them like we handle others. See the investment transaction for example: https://github.com/balancer-labs/frontend-v2/blob/develop/src/components/forms/pool_actions/InvestForm/components/InvestPreviewModal/components/InvestActions.vue#L185-L239

You will need to use a try/catch block to catch any transaction errors, get the pending tx returned from the service and pass it to some kind of tx handler function like in the example which adds the transaction to the global state and uses the txListener to figure out when it's confirmed.